blocklylib-core/src/main/java/com/google/blockly/android/ToolboxFragment.java

```diff
@@ -288,6 +299,12 @@ public void setCurrentCategory(@Nullable ToolboxCategory category) {
         mCategoryTabs.setSelectedCategory(category);
         updateCategoryColors(category);
         mBlockListView.setVisibility(View.VISIBLE);
+        if (category.getCustomType() != null
+                && TextUtils.equals("VARIABLE", category.getCustomType().toUpperCase())) {
```

Comment:
Collapse the check into ToolboxCategory.isVariableCategory(), possibly precomputed at load time.